torch/autograd/graph.py

```diff
@@ -179,7 +179,8 @@ def _get_grad_fn_or_grad_acc(t: Union[torch.Tensor, "GradientEdge"]) -> Node:
     if isinstance(t, GradientEdge):
         return t.node
     if t.requires_grad and t.grad_fn is None:
-        node = t.view_as(t).grad_fn.next_functions[0][0]  # type: ignore[union-attr]
+        with torch.enable_grad():
```

Comment:
note that this line is a non-trivial change and needs to be confirmed by @soulitzer 

This happens here because we register multi-grad hook during the backward where grad_mode is disabled.

Reply:
This change sounds good, though I'm not completely sure I understand why the hooks can be registered when grad_mode is disabled. 
(If it is the checkpoint case, then we should reenable grad before recomputing forward)